src/cloud-api-adaptor/hack/verify-provenance.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# Verify Github's attestation reports. Meant to verify binaries built
+# by upstream projects (kata-containers and guest-components).
+#
+# GH cli is used to verify.
+#
+# Asserts on the claims are:
+# - Triggered by push on the main branch
+# - Built on the given repository
+# - The gh action workflow is matching the given digest
+# - The code is matching the given digest
+#
+# -g will fetch attestation via gh cli, this requires GH_TOKEN to be
```

Comment:
nit: If the description of the "g" option is added to the usage function, then it'll avoid looking at the code for the first time user.